lib/ex_doc/autolink.ex

```diff
@@ -63,28 +63,39 @@ defmodule ExDoc.Autolink do
   end
 
   defp walk({:a, attrs, inner, meta} = ast, config) do
-    cond do
-      url = custom_link(attrs, config) ->
-        {:a, Keyword.put(attrs, :href, url), inner, meta}
+    case custom_link(attrs, config) do
+      :remove_link ->
+        remove_link(ast)
 
-      true ->
+      nil ->
         ast
+
+      url ->
+        {:a, Keyword.put(attrs, :href, url), inner, meta}
     end
   end
 
   defp walk({:code, attrs, [code], meta} = ast, config) do
-    if url = url(code, :regular_link, config) do
-      code = remove_prefix(code)
-      {:a, [href: url], [{:code, attrs, [code], meta}], %{}}
-    else
-      ast
+    case url(code, :regular_link, config) do
+      nil ->
+        ast
+
+      url ->
+        code = remove_prefix(code)
+        {:a, [href: url], [{:code, attrs, [code], meta}], %{}}
```

Comment:
this is not changing anything. It will revert it as it was. It was probably a left over from a different approach I was taking